src/Payments/Integrations/Cheque.php

```diff
@@ -13,56 +13,65 @@
 use Exception;
 use WC_Stripe_Payment_Request;
 use WC_Stripe_Helper;
+use Automattic\WooCommerce\Blocks\Assets\Api;
 
 /**
  * Cheque payment method integration
  *
  * @since $VID:$
  */
-final class Cheque extends AbstractPaymentMethodIntegration {
+final class Cheque extends AbstractPaymentMethodType {
 	/**
-	 * Payment method name.
+	 * Settings from the WP options table
 	 *
-	 * @var string
+	 * @var array
 	 */
-	protected $payment_method_name = 'cheque';
+	private $cheque_settings;
 
 	/**
-	 * Settings from the WP options table
+	 * An instance of the Asset Api
 	 *
-	 * @var array
+	 * @var Api
 	 */
-	private $cheque_settings;
+	private $asset_api;
 
 	/**
-	 * Registers the block type with WordPress.
+	 * Constructor
+	 *
+	 * @param string $name The name of the payment method type.
+	 * @param Api    $asset_api An instance of Api.
 	 */
-	public function register_payment_method() {
+	public function __construct( $name, Api $asset_api ) {
+		$this->asset_api = $asset_api;
+		parent::__construct( $name );
+	}
+
+	/**
+	 * Initializes the payment method type.
+	 */
+	public function initialize() {
 		$this->cheque_settings = get_option( 'woocommerce_cheque_settings', [] );
-		parent::register_payment_method();
 	}
 
 	/**
 	 * Returns an array of scripts/handles to be registered for this payment method.
 	 *
 	 * @return array
 	 */
-	protected function get_payment_method_scripts() {
-		return [
-			[
-				'handle' => 'wc-payment-method-cheque',
-				'src'    => 'build/wc-payment-method-cheque.js',
-				'deps'   => [],
-			],
-		];
+	public function get_payment_method_script_handles() {
+		$this->asset_api->register_script(
+			'wc-payment-method-cheque',
+			'build/wc-payment-method-cheque.js'
+		);
+		return [ 'wc-payment-method-cheque' ];
```

Comment:
As you can see, we leave script/style _registration_ up to the payment methods. Implementing an interface for that in our code seems like it'd be just duplicating what `wp_register_script` already does. So let's just hand that off to payment methods to deal with. All we care about are handles for registered assets.